tests/rule_based_profiler/config/test_base.py

```diff
@@ -0,0 +1,497 @@
+import logging
+
+import pytest
+
+from great_expectations.marshmallow__shade.exceptions import ValidationError
+from great_expectations.rule_based_profiler.config.base import (
+    DomainBuilderConfig,
+    DomainBuilderConfigSchema,
+    ExpectationConfigurationBuilderConfig,
+    ExpectationConfigurationBuilderConfigSchema,
+    ParameterBuilderConfig,
+    ParameterBuilderConfigSchema,
+    RuleBasedProfilerConfig,
+    RuleBasedProfilerConfigSchema,
+    RuleConfig,
+    RuleConfigSchema,
+)
+
+
+def test_domain_builder_config_successfully_loads_with_required_args():
+    data = {
+        "class_name": "DomainBuilder",
+    }
+    schema = DomainBuilderConfigSchema()
+    config = schema.load(data)
+    assert isinstance(config, DomainBuilderConfig)
+    assert all(getattr(config, k) == v for k, v in data.items())
+
```

Comment:
These tests are structured as follows: for each of DomainBuilder, ParameterBuilder, ExpectationConfigurationBuilder, Rule, and RuleBasedProfiler, test both the `load` and `dump` methods achieved through the `Schema` inheritance relationship.

I've tried to separate out scenarios such as required values, optional values, **kwargs, etc to keep tests somewhat simple. I've also captured logging but let me know if you'd like me to move that out to additional tests.

Reply:
we've got the same test repeated because we have the same logic repeated; let's discuss an opportunity here

Reply:
Tests to be cleaned up after refactoring of feature; open to resolve